service/telemetry/config.go

```diff
@@ -122,6 +122,11 @@ type MetricsConfig struct {
 	// Readers allow configuration of metric readers to emit metrics to
 	// any number of supported backends.
 	Readers []MetricReader `mapstructure:"metric_readers"`
+
+	// HostProc is the path to the proc folder used to look up for the collector process.
+	// When unset, the collector will use the environment variable HOST_PROC if set, and
+	// default to `/proc` if no value is set.
```

Comment:
Should this be called out as a unix-only feature? What should happen if it's configured on a windows system? I understand the underlying library will just ignore the value, but it would be good to set expectations for people using the collector.

Should the default value be set somewhere in case the underlying library ever decided to change the default value? Alternatively, a test that would help identify regressions would be useful here

Reply:
This is a Unix specific change, correct. The env vars were only set with Linux, not other OSes.

Reply:
We also set the default values ourselves always, and in any case rely on gopsutil otherwise. Do you want me to call out that this config entry is only applicable to Linux?

Reply:
Would it make sense to have this configuration option only available on Linux via build tags? Something like:

```go
type MetricsConfig struct {
  // Existing fields go here
  // ...

  // OSSpecificMetricsConfig defines OS-specific metrics configuration
  OSSpecificMetricsConfig `mapstructure:",squash"`
}
```

and then we have different definitions of the `OSSpecificMetricsConfig` struct for Linux (having the `HostProc` setting) and Windows/not Linux (where `OSSpecificMetricsConfig` would just be an empty struct). 

This makes the option available only on Linux for both library users and end-users of the binary, and implicitly documents that the feature is Linux-only.

Reply:
I don't know which would be more confusing to users, to have an error when using the same config across OS's or to apply the same config and have one of the options do nothing on some platforms.

I would expect the behaviour to be the same as if someone tried to use config that isn't available on certain platforms in the hostmetrics receiver

Reply:
Agree that following precedent on hostmetricsreceiver makes sense if we already have OS-specific options. If we are not sure which path to follow, I think raising an error is better in that we can always change our minds if people complain, while we can't easily do that with the "ignore" behavior.

Reply:
> I am a bit hesitant about this, would it break the collector to run on Windows with a message saying the config is unrecognized? I can log a warning on Windows if you like, but I'm not sure we should use different configs as it might confuse users.

Yes, it would fail on Windows. I would at least raise a warning. A benefit of failing on Windows is that we can change that behavior later, while if we commit to a specific behavior on Windows it will be hard to change later

Reply:
How about I cut this PR in 2 and first introduce this behavior, and then a separate PR to discuss how we expose this in config?

Reply:
@atoulme makes sense to me!

Reply:
Thanks, updated the PR.